qadence/constructors/hamiltonians.py

```diff
@@ -119,9 +119,19 @@ def hamiltonian_factory(
 
     # Get interaction function
     if interaction is not None:
-        int_fn = INTERACTION_DICT.get(interaction, None)
-        if int_fn is None:
-            raise KeyError(f"Interaction {interaction} not supported.")
+        if callable(interaction):
+            int_fn = interaction
+            try:
```

Comment:
Is the try/catch block still needed ? It feels you can get away with it now.

Reply:
It's still relevant in case the call to `interaction(0,1)` fails due to an incorrect signature.